packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -619,6 +619,17 @@ export default class BackupNg {
         const jobId = job.id
 
         const remoteIds = unboxIdsFromPattern(job.remotes)
+        for (const id of remoteIds) {
+          const remote = await app.getRemote(id)
+          if (remote.proxy !== job.proxy) {
+            throw new Error(
+              job.proxy !== undefined
+                ? `The remote ${remote.name} should be linked to the proxy ${job.proxy}`
+                : `The remote ${remote.name} should not be linked to a proxy`
+            )
+          }
+        }
```

Comment:
You can parallelize.

Reply:
I don't think is a good solution, parallelism will execute 'n' promise. This implementation will execute '1...n' promise.

Reply:
@badrAZ Your explanation is not very clear, I had to read a few times to understand and I'm not even sure :stuck_out_tongue: 

First of all, a promise is not executed, it's the result of an async operation.

Do you mean that your implementation will not need to check all remotes because it will fail on the first one?

If that's the case, I think it's not a good optimization, because the error should not happen, therefore this case is very unlikely and you will always check all remotes but sequentially instead of in parallel.